### PR TITLE
fix(drivers): name a location when the missing-driver error searched nothing

### LINKED ISSUE #1191: Missing-driver error says "Searched 0 locations:" and names nowhere on a first run
Drivers are deliberately not shipped with the binary, so "driver not installed" is a normal, expected state that users hit by design. That makes the message a primary UX surface — and in the first-run case it currently names nowhere to look.

`driverSearchRoots()` only returns directories that already exist. On a machine that has never installed a driver, every candidate root is absent, so it returns an empty array and `DriverNotInstalledError` renders:

```
DuckDB driver not installed.
Install it with the warehouse_install_driver tool, or run:
  npm install --prefix /home/user/.local/share/altimate-code/drivers duckdb
Searched 0 locations:
```

The last line ends in a colon with nothing after it. It reads like a truncated message or a bug in the CLI rather than the ordinary "you have not installed this yet" state, and it names no location at all.

Expected: the message should say plainly that there was nothing to search, and name the directory the install command creates, so the two failure modes stay distinguishable:
- not installed anywhere (this case)
- found on disk but failed to load (already handled separately, and already clear)

Reproduce with a compiled binary and a driver directory that does not exist:

```
ALTIMATE_DRIVER_DIR=/tmp/does-not-exist altimate ...  # any DuckDB operation
```

Scope: text and one branch in `DriverNotInstalledError` in `packages/drivers/src/resolve.ts`. No change to resolution behaviour.

### PR DESCRIPTION
**Pull request by anandgupta42**
### Issue for this PR

Closes #1191.

This PR targets `fix/warehouse-driver-bootstrap` (#1122), not `main`; it builds on the optional-driver resolver introduced by that PR and includes the latest parent head.

### Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Refactor / code improvement
- [ ] Documentation

### What does this PR do?

When optional-driver resolution fails and `driverSearchRoots()` returns an empty list, the old error ended with a dangling `Searched 0 locations:` line. The first version of this PR improved that line but overclaimed that the driver directory itself did not exist.

The final wording states only what the empty list proves and still gives the user an actionable destination:

```text
DuckDB driver not installed.
Install it with the warehouse_install_driver tool, or run:
  npm install --prefix /home/user/.local/share/altimate-code/drivers duckdb
No searchable driver locations were found. Expected managed location: /home/user/.local/share/altimate-code/drivers/node_modules.
```

When roots exist, the existing `Searched N location(s): ...` message is unchanged. Resolution, root precedence, package selection, dynamic import, installation, and permissions are unchanged. In particular, the parent branch deliberately excludes project and ancestor `node_modules`; it retains the consent-gated managed root and explicitly configured/runtime roots.

### Review feedback addressed

The empty-list regression test explicitly models the constructor boundary. Forcing `driverSearchRoots()` to return `[]` in a unit-test checkout would be host-dependent because executable- or package-adjacent roots may legitimately exist.

A late review found the complementary portability problem in the older non-empty-root test: it asserted `Searched` without creating any root, so it could fail in a dependency-free checkout. The final test-only commit creates its own managed `node_modules` root, points `ALTIMATE_DRIVER_DIR` at its parent, and asserts that exact root. The direct-constructor test still covers the genuine empty-root wording.

Codebase graph tracing confirms that `loadOptionalDriver()` passes the exact `driverSearchRoots()` array to `DriverNotInstalledError` without transformation.

### Verification

| Gate | Result |
|---|---|
| Focused resolver tests | 68 pass, 0 fail |
| Combined `packages/drivers` full suite | 225 pass, 0 fail |
| Driver catalogue consistency | 7 pass, 0 fail |
| `bun run typecheck` | 13/13 successful |
| `git diff --check` | pass |
| Three-seat Council consensus review | unanimous ship-after-fix, high confidence |
| Codex Security exact-final-head diff scan | complete coverage, 0 reportable findings |

The source branch was synchronized with final #1122 head `c49149a26f` using a normal merge. Final reviewed/pushed head: `212f4546a7`. No rebase or force-push was used.

### Screenshots / recordings

Not a UI change.

### Checklist

- [x] I have tested my changes locally
- [x] I have not included unrelated changes in this PR

---

> [!NOTE]
> **Low Risk**
> User-facing error text only in the drivers package; no changes to resolution, install paths, or security-sensitive logic.
> 
> **Overview**
> When optional driver resolution fails and **`driverSearchRoots()`** returns no existing directories (common on a first compiled run), **`DriverNotInstalledError`** no longer ends with a useless **`Searched 0 locations:`** line. It now says that no searchable locations were found and points at the **expected managed `node_modules` path** under **`driverInstallDir()`**, while keeping the existing **`Searched N location(s): …`** wording when roots were actually checked.
> 
> The **`npm install --prefix`** hint still uses the same install directory (now via a single **`installDir`** variable). Resolver behavior is unchanged.
> 
> Tests cover the empty-**`searched`** message directly (host-independent), assert listed roots when **`searched`** is non-empty, and tighten the integration case so the error mentions the managed root.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 212f4546a70255155f5efb4284517f3ab4d2271a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/drivers/test/resolve-unit.test.ts [hunk 1]
```diff
@@ -608,6 +608,31 @@ describe("manual-install hints are copy-pasteable", () => {
     expect(prefix!.startsWith("'") || prefix!.startsWith('"')).toBe(true)
   })
 
+  test("DriverNotInstalledError names a location when nothing was searched", () => {
+    process.env["ALTIMATE_DRIVER_DIR"] = path.join(path.sep, "tmp", "altimate-drivers-absent")
+
+    // The first-run state: no driver directory exists, so driverSearchRoots()
+    // returns nothing. The message must still name somewhere.
+    const err = new DriverNotInstalledError("duckdb", DRIVER_PACKAGES.duckdb, [])
```
[the remark this answers is not in the document]
Comment (anandgupta42) [reply packages/drivers/test/resolve-unit.test.ts:839]: Validated the production path before changing this: `loadOptionalDriver()` computes `const roots = driverSearchRoots()` and passes that exact array directly to `DriverNotInstalledError`. The hard-coded `[]` is therefore intentionally the constructor-boundary case; forcing the process-wide resolver to have no roots is host-dependent because executable/package-adjacent roots can legitimately exist in a test checkout.

The comment was right that the old test explanation overclaimed the modeled state, and the production wording also overclaimed that the managed directory itself was absent. Commit 31a73e1493 fixes both: the test now names the empty-list boundary and uses a guaranteed per-test path, while the message says only that no searchable locations were found and labels the managed path as expected. Focused resolver tests pass 68/68; full drivers pass 217/217.

File: packages/drivers/src/resolve.ts [hunk 2]
```diff
@@ -91,11 +91,19 @@ export class DriverNotInstalledError extends Error {
...
   constructor(driver: DriverName, packages: readonly string[], searched: readonly string[]) {
     const label = DRIVER_LABELS[driver]
+    const installDir = driverInstallDir()
+    // `driverSearchRoots()` only returns directories that exist, so a compiled
+    // first run can have no searchable roots at all. The old message then ended
+    // in a bare "Searched 0 locations:" with nothing after the colon. Describe
+    // only what the empty list proves and name the managed location users need.
+    const searchedLine = searched.length
+      ? `Searched ${searched.length} location${searched.length === 1 ? "" : "s"}: ${searched.join(", ")}`
+      : `No searchable driver locations were found. Expected managed location: ${path.join(installDir, "node_modules")}.`
```
[the remark this answers is not in the document]
Comment (anandgupta42) [reply packages/drivers/src/resolve.ts:101]: Fixed in 212f4546a7. The older searched-root test now creates its own managed node_modules root, points ALTIMATE_DRIVER_DIR at its parent, and asserts that exact root, so it deterministically exercises the non-empty `Searched` branch even in a dependency-free checkout. The direct-constructor test still covers the genuine empty-root wording. Verified with 68/68 resolver tests, 225/225 driver tests, and 13/13 typecheck tasks.
